llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -1579,6 +1579,12 @@ RISCVInstrInfo::isCopyInstrImpl(const MachineInstr &MI) const {
   switch (MI.getOpcode()) {
   default:
     break;
+  case RISCV::ADD:
+    if (MI.getOperand(1).isReg() && MI.getOperand(1).getReg() == RISCV::X0)
+      return DestSourcePair{MI.getOperand(0), MI.getOperand(2)};
```

Comment:
Do we need to check that Operand 2 is a register?

Reply:
Isn't that implied by the instruction opcode?  It can't be an immediate.  Are you thinking a frame index or something like that?

Reply:
Yes, though that's probably impossible given current FrameIndex usage.